contrib/inventory/ec2.py

```diff
@@ -1479,9 +1166,50 @@ def get_instance_route53_names(self, instance):
 
     def get_host_info_dict_from_instance(self, instance):
         instance_vars = {}
-        for key in vars(instance):
-            value = getattr(instance, key)
-            key = self.to_safe('ec2_' + key)
+
+        hostvars = ['ami_launch_index',
+                    'architecture',
+                    'block_device_mappings',
+                    'client_token',
+                    'ebs_optimized',
+                    'elastic_gpu_associations',
+                    'ena_support',
+                    'hypervisor',
+                    'iam_instance_profile',
+                    'id',
+                    'image_id',
+                    'instance_lifecycle',
+                    'kernel_id',
+                    'key_name',
+                    'launch_time',
+                    'monitoring',
+                    'network_interfaces_attribute',
+                    'placement',
+                    'placement_group',
+                    'platform',
+                    'private_dns_name',
+                    'private_ip_address',
+                    'product_codes',
+                    'public_dns_name',
+                    'public_ip_address',
+                    'ramdisk_id',
+                    'root_device_name',
+                    'root_device_type',
+                    'security_groups',
+                    'source_dest_check',
+                    'spot_instance_request_id',
+                    'sriov_net_support',
+                    'state',
+                    'state_reason',
+                    'state_transition_reason',
+                    'subnet_id',
+                    'tags',
+                    'virtualization_type',
+                    'vpc_id']
+
+        for var in hostvars:
+            value = getattr(instance, var)
```

Comment:
I haven't looked too closely into it, but I ran into an issue where `elastic_gpu_associations` wasn't defined for some of my instances so this getattr call would fail. I used this as a work around:

``` python
            value = getattr(instance, var, None)
            if value is None:
                continue
```